### PR TITLE
[clang-doc] Support markdown and simplify checks

### PR DESCRIPTION
**Pull request by PeterChou1**
This patch modifies the basic-project in clang-doc. Currently we're matching the entire html output. This patch modifies it so that we only match the parts relevant to the documentation logic instead just matching the boilerplate code. This patch also adds the markdown output to the basic-project test

### CODE REVIEW

File: clang-tools-extra/test/clang-doc/basic-project.test [hunk 1]
```diff
@@ -54,306 +54,249 @@
...
-// HTML-SHAPE-NEXT:           </li>
-// HTML-SHAPE-NEXT:         </ul>
-// HTML-SHAPE-NEXT:       </li>
-// HTML-SHAPE-NEXT:     </ol>
-// HTML-SHAPE-NEXT:   </div>
-// HTML-SHAPE-NEXT: </main>
+// HTML-SHAPE: <h1>class Shape</h1>
+// HTML-SHAPE: <p>Defined at line 8 of file {{.*}}Shape.h</p>
+// HTML-SHAPE: <div>
+// HTML-SHAPE:   <div>
```
Comment (ilovepi) [inline clang-tools-extra/test/clang-doc/basic-project.test:60]: Do you need to match these? are they important? same w/ all the lines that only have tags...

**Review by ilovepi — COMMENTED** [review]
This is headed in the right direction, but I'm not sure we want to match so many of the single line tags ... they don't seem particularly valuable from a testing standpoint. If you think we're missing tests for the HTML structure, then we should have some tests that focus on just that aspect that can be a bit simpler in nature than a full project.

Ultimately, I'd like to see a subset of tests that veryify that we do the documentation correctly, and a set of tests that exercise the format enough to be sure we're emiting well structured markdown/yaml/html.

File: clang-tools-extra/test/clang-doc/basic-project.test (hunk 1 above)
Comment (PeterChou1) [reply clang-tools-extra/test/clang-doc/basic-project.test]: good point I remove everything except for relevant infromation

**Review by ilovepi — APPROVED** [review]
LGTM, but let’s update the pr title and description to match the current patch. 

Also, a title like modify tests isn’t very informative. Something like “Support markdown and simplify checks” seems more appropriate. Additionally, those two aspects seem like they could be separate PRs, though I won’t require that.

**Pull request merged**